apps/ftu/test/unit/sim_manager_test.js

```diff
@@ -582,4 +582,46 @@ suite('sim mgmt >', function() {
     });
   });
 
+  suite('SIM2 inserted >', function() {
+    var iccId1;
+    var iccInfo1;
+    var realIccId0;
+
+    suiteSetup(function() {
+      iccId1 = '98765';
+      navigator.mozIccManager.addIcc(iccId1);
+      iccInfo1 = navigator.mozIccManager.getIccById(iccId1);
+      navigator.mozMobileConnections.mAddMobileConnection();
+      navigator.mozMobileConnections[1].iccId = iccId1;
+      navigator.mozMobileConnections[1].iccInfo = iccInfo1;
+
+      SimManager.updateIccState(iccId1);
+      SimManager.simSlots = 2;
+
+      realIccId0 = SimManager.icc0;
+      SimManager.icc0 = null;
+    });
+
+    suiteTeardown(function() {
+      navigator.mozIccManager.removeIcc(iccId1);
+      navigator.mozMobileConnections.mRemoveMobileConnection(1);
+      SimManager.simSlots = 1;
+      SimManager.icc1 = null;
+      SimManager.icc0 = realIccId0;
+    });
+
+    test('"simUnlockBack" hides the screen', function() {
+      iccInfo1.cardState = 'pinRequired';
+      SimManager.handleCardState();
+
+      navigationStub = sinon.stub(Navigation, 'back');
+      SimManager.simUnlockBack();
+      assert.ok(navigationStub.calledOnce);
+      assert.isTrue(UIManager.activationScreen.classList.contains('show'));
+      assert.isFalse(UIManager.unlockSimScreen.classList.contains('show'));
+      assert.isTrue(UIManager.pinRetriesLeft.classList.contains('hidden'));
+      navigationStub.restore();
+    });
+  });
+
```

Comment:
we should also add a test checking that:
- SIM1 is PIN protected, but we skip it
- SIM2 is PIN protected, but we go back
- then SIM1 PIN screen is shown again (we are testing for hidden right now)